runtime/lua/vim/lsp/util.lua

```diff
@@ -423,27 +423,58 @@ function M.make_floating_popup_options(width, height, opts)
   }
 end
 
+local function is_new_item(item)
+  local stack = gettagstack()
+
+  -- Check if we're at the bottom of the tagstack.
+  if stack.curidx <= 1 then return true end
+
+  local top_item = stack.items[stack.curidx-1]
+
+  -- Check if the item at the top of the tagstack is exactly the
+  -- same as the one we want to push.
+  if top_item.tagname ~= item.tagname then return true end
+  for i, v in ipairs(top_item.from) do
+    if v ~= item.from[i] then return true end
+  end
+  return false
+end
+
+local function add_tagstack(item)
```

Comment:
should be private :
```suggestion
local function _add_tagstack(item)
```

Reply:
Thanks for the tips, I'm new to Lua, so it's nice to learn from you all.